cmake/modules/CheckCompileTargets.cmake

```diff
@@ -0,0 +1,70 @@
+# CMake module to get the compile options to match the target 
```

Comment:
@tone-zhang i'd suggest you have two commits for this PR

1. a refactor only commit: extract `CheckCompileTargets.cmake` out of `src/CMakeLists.txt` and use this module in `src/jerssure/CMakeLists.txt` instead of repeating it in the top level `CMakeLists.txt`. i think it's better off include the module at the place where we are using it.
1. a bug fix for AArch64 build: to `include(CheckCompileTargets)` in `src/test/CMakeLists.txt`. and other fixes for building failures on ARM.

also 

1. could you fix the typo in your commit message: s/AArm64/AArch64/ ?
1. could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#preparing-and-sending-patches. in this case, the prefix would be "cmake: ".